packages/devextreme/testing/tests/DevExpress.ui.widgets/speechToText.tests.js

```diff
@@ -0,0 +1,619 @@
+            enabled: false,
+            isListening: false,
+        });
+
+        assert.ok(!this.$element.hasClass(SPEECH_TO_TEXT_LISTENING_CLASS), 'reset to initial when custom engine disabled');
+
+        const $button = this.getButton();
+        $button.trigger('dxclick');
+
+        assert.ok(this.$element.hasClass(SPEECH_TO_TEXT_LISTENING_CLASS), 'manual control works');
+    });
+
+    QUnit.test('should handle state transitions with disabled component', function(assert) {
+        this.reinit({ disabled: true });
+
+        const $button = this.getButton();
+        $button.trigger('dxclick');
+
+        assert.ok(!this.$element.hasClass(SPEECH_TO_TEXT_LISTENING_CLASS), 'disabled prevents state change');
+
+        this.instance.option('disabled', false);
+        this.instance.option('customSpeechRecognizer', {
+            enabled: true,
+            isListening: true,
+        });
+
+        assert.ok(this.$element.hasClass(SPEECH_TO_TEXT_LISTENING_CLASS), 'custom engine works after enabling');
+    });
+});
+
+QUnit.module('Events', moduleConfig, () => {
+    QUnit.test('onStartClick should be triggered on first button click', function(assert) {
+        const onStartClickSpy = sinon.spy();
+
+        this.reinit({ onStartClick: onStartClickSpy });
+
+        this.getButton().trigger('dxclick');
+
+        assert.ok(onStartClickSpy.calledOnce, 'onStartClick event fired exactly once');
+    });
+
+    QUnit.test('onStopClick should be triggered on second button click', function(assert) {
+        const onStopClickSpy = sinon.spy();
+
+        this.reinit({ onStopClick: onStopClickSpy });
+
+        const $button = this.getButton();
+
+        $button.trigger('dxclick');
+        $button.trigger('dxclick');
+
+        assert.ok(onStopClickSpy.calledOnce, 'onStopClick event fired exactly once');
+    });
+
+    QUnit.test('should pass correct event arguments to onStartClick', function(assert) {
+        assert.expect(4);
+
+        this.reinit({
+            onStartClick: (e) => {
+                assert.ok(e.component instanceof SpeechToText, 'component is correct');
+                assert.strictEqual(e.element, this.$element.get(0), 'element is provided correctly');
+                assert.ok(e.event, 'original event is provided');
+                assert.strictEqual(typeof e.event, 'object', 'event is an object');
+            },
+        });
+
+        this.getButton().trigger('dxclick');
+    });
+
+    QUnit.test('should pass correct event arguments to onStopClick', function(assert) {
+        assert.expect(4);
+
+        this.reinit({
+            onStopClick: (e) => {
+                assert.ok(e.component instanceof SpeechToText, 'component is correct');
+                assert.strictEqual(e.element, this.$element.get(0), 'element is provided correctly');
+                assert.ok(e.event, 'original event is provided');
+                assert.strictEqual(typeof e.event, 'object', 'event is an object');
```

Comment:
Test would be more strict if you trigger not just a click but trigger an event directly `$.Event('click')` - then there will be a single assert that event is passed correctly